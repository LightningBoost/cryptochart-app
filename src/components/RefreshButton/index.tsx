import React, {useEffect} from 'react';
import {TouchableOpacity, Animated, Easing} from 'react-native';
import FontAwesome5Icon from 'react-native-vector-icons/FontAwesome';
import {useTheme} from 'react-native-paper';

FontAwesome5Icon.loadFont();

interface Props {
  onPress: () => void;
  size?: number;
  spin?: boolean;
}

const RefreshButton: React.FC<Props> = ({onPress, spin = false, size = 18}) => {
  const theme = useTheme();
  const spinValue = new Animated.Value(0);

  const spinFunction = () => {
    spinValue.setValue(0);

    Animated.timing(spinValue, {
      toValue: 1,
      duration: 1000,
      easing: Easing.linear,
      useNativeDriver: true,
    }).start(() => {
      if (spin) {
        spinFunction();
      }
    });
  };

  const rotate = spinValue.interpolate({
    inputRange: [0, 1],
    outputRange: ['0deg', '360deg'],
  });

  useEffect(() => {
    if (spin) {
      spinFunction();
    }
    // eslint-disable-next-line
  }, [spin]);

  return (
    <TouchableOpacity onPress={onPress}>
      <Animated.View style={{transform: [{rotate}]}}>
        <FontAwesome5Icon
          name="refresh"
          size={size}
          color={theme.colors.text}
        />
      </Animated.View>
    </TouchableOpacity>
  );
};

export default RefreshButton;
