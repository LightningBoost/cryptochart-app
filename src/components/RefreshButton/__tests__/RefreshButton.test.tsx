import React from 'react';
import renderer from 'react-test-renderer';
import RefreshButton from '../index';

it('should match the snapshot', () => {
  const mockFn = jest.fn();

  const tree = renderer.create(<RefreshButton onPress={mockFn} />).toJSON();

  expect(tree).toMatchSnapshot();
});
