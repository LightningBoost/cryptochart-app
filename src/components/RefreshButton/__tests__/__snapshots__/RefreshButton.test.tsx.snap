// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "transform": Array [
          Object {
            "rotate": "0deg",
          },
        ],
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontSize": 18,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      ï€¡
    </Text>
  </View>
</View>
`;
